graphql-dgs-subscriptions-sse/src/main/kotlin/com/netflix/graphql/dgs/subscriptions/sse/DgsSSESubscriptionHandler.kt

```diff
@@ -44,119 +49,74 @@ import java.util.*
-            emitter.complete()
-            return ResponseEntity.badRequest().body(emitter)
+            throw ServerWebInputException("Error parsing query: ${ex.message}")
+        }
+
+        if (!isSubscriptionQuery(queryPayload.query)) {
+            throw ServerWebInputException("Invalid query. operation type is not a subscription")
         }
 
         val executionResult: ExecutionResult = dgsQueryExecutor.execute(queryPayload.query, queryPayload.variables)
         if (executionResult.errors.isNotEmpty()) {
-            return if (
-                executionResult.errors.asSequence().filterIsInstance<ValidationError>().any() ||
-                executionResult.errors.asSequence().filterIsInstance<InvalidSyntaxError>().any()
-            ) {
-                val errorMessage = "Subscription query failed to validate: ${executionResult.errors.joinToString(", ")}"
-                emitter.send(errorMessage)
-                emitter.complete()
-                ResponseEntity.badRequest().body(emitter)
+            val errorMessage = if (executionResult.errors.any { error -> error is ValidationError || error is InvalidSyntaxError }) {
+                "Subscription query failed to validate: ${executionResult.errors.joinToString()}"
             } else {
-                val errorMessage = "Error executing subscription query: ${executionResult.errors.joinToString(", ")}"
-                logger.error(errorMessage)
-                emitter.send(errorMessage)
-                emitter.complete()
-                ResponseEntity.status(500).body(emitter)
+                "Error executing subscription query: ${executionResult.errors.joinToString()}"
             }
-        }
-
-        val subscriber = object : Subscriber<ExecutionResult> {
-            lateinit var subscription: Subscription
-
-            override fun onSubscribe(s: Subscription) {
-                logger.info("Started subscription with id {} for request {}", sessionId, queryPayload)
-                subscription = s
-                s.request(1)
-            }
-
-            override fun onNext(t: ExecutionResult) {
-                val event = SseEmitter.event()
-                    .data(
-                        mapper.writeValueAsString(
-                            SSEDataPayload(data = t.getData(), errors = t.errors, subId = sessionId)
-                        ),
-                        MediaType.APPLICATION_JSON
-                    ).id(UUID.randomUUID().toString())
-                emitter.send(event)
-
-                subscription.request(1)
-            }
-
-            override fun onError(t: Throwable) {
-                logger.error("Error on subscription {}", sessionId, t)
-                val event = SseEmitter.event()
-                    .data(
-                        mapper.writeValueAsString(
-                            SSEDataPayload(
-                                data = null,
-                                errors = listOf(Error(t.message)),
```

Comment:
I think this line was actually a bug. `Error` here is actually a typealias for `java.lang.Error`, so we'd end up sending back a full stack trace. I changed it to use `com.netflix.graphql.types.subscription.Error`, which just has a `message` field instead, since I think that is what was intended. Maybe the name of the class should be changed so that it doesn't so easily conflict with the kotlin `Error` typealias?